ShowerShow/Control/ShowerController.cs

```diff
@@ -11,45 +11,62 @@
 using System;
 using ShowerShow.DTO;
 using ShowerShow.Models;
+using System.Collections.Generic;
+using ShowerShow.Repository.Interface;
+using ShowerShow.Service;
 
 namespace ShowerShow.Controllers
 {
     public class ShowerController
     {
         private readonly ILogger<ShowerController> _logger;
+        private readonly IShowerService _showerService;
 
-        public ShowerController(ILogger<ShowerController> log)
+        public ShowerController(ILogger<ShowerController> log, ShowerService showerService)
         {
             _logger = log;
+            _showerService = showerService;
         }
-        [Function("CreateShower")]
-        [OpenApiOperation(operationId: "CreateShower", tags: new[] { "Showers" })]
-        [OpenApiRequestBody("application/json", typeof(CreateShowerDataDTO), Description = "The shower data.")]
+
+        [Function(nameof(GetShowerById))]
+        [OpenApiOperation(operationId: "getShowerById", tags: new[] { "Shower data" })]
+        [OpenApiParameter(name: "UserId", In = ParameterLocation.Path, Required = true, Type = typeof(Guid), Description = "The user id parameter")]
+        [OpenApiParameter(name: "ShowerId", In = ParameterLocation.Path, Required = true, Type = typeof(Guid), Description = "id of the requested shower")]
+        [OpenApiResponseWithBody(statusCode: HttpStatusCode.OK, contentType: "application/json", bodyType: typeof(ShowerData), Description = "Successfully received the shower data.")]
+        public async Task<HttpResponseData> GetShowerById([HttpTrigger(AuthorizationLevel.Anonymous, "get", Route = "user/{UserId}/showerdata/{showerId}")] HttpRequestData req, Guid UserId, Guid showerId)
+        {
+
+            var response = _showerService.GetShowerDataByUserId(UserId, showerId);
```

Comment:
I have added thje method checkif user exist. But also consider the fact about shower id might not exist in db. You will need the same checkif method for shower id and do some error handling here 👍

Reply:
![image](https://user-images.githubusercontent.com/54850459/195910117-2c807148-f3a8-490f-b625-b31ba8f68a04.png)